selftests/check.py

```diff
@@ -539,9 +553,12 @@ def main():
     # Job execution
     # ========================================================================
     config = {'core.show': ['app'],
-              'run.test_runner': 'nrunner'}
+              'run.test_runner': 'nrunner',
+              'run.journal.enabled': True}
     with Job(config, suites) as j:
-        return j.run()
+        exit_code = j.run()
+    print_failed_tests(j.logdir)
+    return exit_code
```

Comment:
For instance: I'm trying to understand why having a journal dependency and the SQL code is better than this:

```python
for test in job.result.tests: 
    if test.get('status') in ['FAIL', 'ERROR']: 
        print(test.get('name'), test.get('status')) 
```

Probably I'm missing something.

Reply:
There's little structure to the individual `test` instances.  A believe a task in itself is to document what users should expect to find here (and maybe even convert those from dicts into more structured objects).  Like I said elsewhere, I believe this a larger (but good) task.

Reply:
I didn't understand your first sentence.

I agree that we should document what users should expect from our API, but I missing the point to reject this solution and stick with the current one, which is getting even deeper, looking at table structures on a SQL database on "tests" results. Not sure if I understood why this more suitable than querying those Python attributes.

Like I said I will not reject this PR, of course, but I still find strange have this SQL code here.

Reply:
> I didn't understand your first sentence.
> 
> I agree that we should document what users should expect from our API, but I missing the point to reject this solution and stick with the current one, which is getting even deeper, looking at table structures on a SQL database on "tests" results. Not sure if I understood why this is more suitable than querying those Python attributes.
> 

What I mean was that each `test` instance is a dictionary, with keys that well, users can not currently have many expectations about.  The upside of the journal file, is that it's more structured.

Now, we all agree that in the end, we want a nice Python API for the results of a job (and tests).

If you've tried the code snippet you proposed, and that's that, then maybe @richtja will be able to quickly come up with a new version.  What I wouldn't like is for this to be delayed waiting for an alternative implementation.

> Like I said I will not reject this PR, of course, but I still find strange have this SQL code here.

Reply:
So we are on the same page here. I tried the code in an isolated job API usage. Not inside this code. 

@richtja is up to you, sr.